poc/dynamic-renderer.php

```diff
@@ -13,14 +13,13 @@ public function registerContentType($type, Callable $handler) {
 	public function __call($method, $arguments) {
 		if (preg_match('~^visit(.+)~', $method, $match)) {
 			$type = strtolower($match[1]);
-			if (!isset($this->handlers[$type])) {
-				$type = 'default';
-			}
 			if (isset($this->handlers[$type])) {
 				$handler = $this->handlers[$type];
 				$this->output .= $handler($arguments[0]);
-			} else {
-				echo "\nLogWarn: Unknown content type {$match[1]}, no default\n";
+                        } 
+                        else {
+                                $type = 'default';
```

Comment:
Does not make sense to switch to default, as it does not exist at that point. In production code, an Exception is thrown instead of the echo.